app/services/filter_application_choices_for_providers.rb

```diff
@@ -2,12 +2,40 @@ class FilterApplicationChoicesForProviders
   def self.call(application_choices:, filters:)
     return application_choices if filters.empty?
 
-    if filters[:status] && filters[:provider]
-      application_choices.where(status: filters[:status].keys, 'courses.provider_id' => filters[:provider].keys)
-    elsif filters[:status]
+    create_filter_query(application_choices, filters)
+  end
+
+  class << self
+  private
+
+    def prepare_search_array(search_terms)
+      search_terms.downcase.gsub(/\W /, '').split
+    end
+
+    def search(application_choices, candidates_name)
+      search_array = prepare_search_array(candidates_name)
+      application_choices.where('first_name ILIKE ANY (array[?])', search_array)
+        .or(application_choices.where('last_name ILIKE ANY (array[?])', search_array))
+    end
```

Comment:
Had a mess around with `psql` after we spoke just now. This seems to work quite well:

```suggestion
    def search(application_choices, candidates_name)
      application_choices.where("CONCAT(first_name, ' ', last_name) ILIKE ?", "%#{candidates_name}%")
    end
```

Reply:
Nice much cleaner - will update to this 👍